src/clarity-angular/datagrid/datagrid.ts

```diff
@@ -116,6 +116,10 @@ export class Datagrid implements AfterContentInit, AfterViewInit, OnDestroy {
         this.refresh.emit(state);
     }
 
+    public hasFilters(): boolean {
+        return this.filters.getActiveFilters().length > 0 ? true : false;
```

Comment:
Active filters are just the ones which are currently filtering something. For instance, a string filter with an empty input (the user hasn't typed anything yet) isn't active, so this will return `false` and you'll get a jacked up display even though the user can open the filter.